stustison/ddf
# Add IdP Configuration screen to installer

Adds a screen to the installer for configuring an IdP.

Screenshots:
![image](https://user-images.githubusercontent.com/14284361/57311526-aa8daa80-70a0-11e9-9a1e-4bc14655081b.png)
![image](https://user-images.githubusercontent.com/14284361/57311705-035d4300-70a1-11e9-9510-1ec8e868416e.png)

Test steps:
  1. Build, unzip, and start this branch of DDF.
  1. When karaf starts, verify that running `lna` in the karaf console prints out only expected bundles.
  1. Open `http://localhost:8993/admin` in your browser, and login as `admin:admin` to begin installation.
  1. At the `Welcome` page, click `Start` to continue.
  1. At the `Setup Types` page, select `Standard` and click next. It should take multiple minutes for this step to complete.
1. At the `IdP Configuration` page:
    1. On the `SAML` tab:
        1. Set `IdP Metadata` to `https://localhost:8993/services/idp/login/metadata`.
        1. Check `Perform User-Agent Check`
        1. Set `SP Metadata` to `https://localhost:8993/services/saml/sso/metadata`, `test`, and `testDelete`. Delete `testDelete`.
        1. Uncheck `Require Sigend AuthnRequests`.
        1. Uncheck `Limit RelatStates to 80 Bytes`.
        1. Set `Cookie Expiration Time (minutes)` to `ten`.
        1. Check `Allow Guest Access`.
    1. On the `OIDC` tab:
        1. Set `IdP Type` to `Generic`.
        1. Set `Client ID` to `ddf.client test`.
        1. Set `Realm/Tenant` to `master test`.
        1. Set `Secret` to `secret test`.
        1. Set `Discovery URI` to `http://localhost:8080/auth/realms/master/.well-known/openid-configuration test`.
        1. Set `Base URI` to `http://localhost:8080/auth test`.
        1. Set `Logout URI` to `http://localhost:8080/auth/admin/master/protocol/openid-connect/logout test`
        1. Set ` Scope` to `openid profile email resource.read test`.
        1. Uncheck `Use Nonce`.
        1. Set `Flow` to `code id_token token`.
        1. Set `Mode` to `form_post`.
1. Click on the `Next ->` button. An error should pop up above the progress bar saying "There is an error in one or more field(s). Please correct and try again.".
1. Verify that there is an error message under the `SAML` tab, below the `Cookie Expiration Time (minutes)` field. Verify that changing the value to `10` makes the error message go away.
1. Click on the `Next ->` button.
1. Click on the `<- Previous` button and verify that the fields are still set on their new values under both tabs.
1. Click on the `Next ->` button.
1. Click on the `Next ->` button.
1. Click on the `Restart Now` button.
1. When karaf starts, verify that running `lna` in the karaf console prints out only expected bundles.
1. Log into the admin console.
1. Verify that under `System -> Configuration`:
    1. `IdP Client` has the configured values.
    1. `IdP Server` has the configured values.
    1. `OIDC Configuration Settings` has the configured values.
